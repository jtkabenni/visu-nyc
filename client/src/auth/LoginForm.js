import React, { useState } from "react";
import { useNavigate } from "react-router-dom";
// import Auth0test from "./Auth0test";
/** Form for creating a new snack or drink item.
 *
 * Has state for the name/quantity of the item; on submission,
 * sends {name, qty} to fn rec'd from parent.
 *
 */

const LoginForm = ({ login }) => {
  const INITIAL_STATE = {
    username: "",
    password: "",
  };
  const navigate = useNavigate();
  const [formData, setFormData] = useState(INITIAL_STATE);
  const [formErrors, setFormErrors] = useState([]);

  async function handleSubmit(evt) {
    evt.preventDefault();
    let result = await login(formData);
    if (result.success) {
      navigate("/");
    } else {
      setFormErrors(result.errors);
    }
  }

  /** Update local state w/curr state of input elem */

  const handleChange = (evt) => {
    const { name, value } = evt.target;
    setFormData((fData) => ({
      ...fData,
      [name]: value,
    }));
  };

  /** render form */

  return (
    <div className="form-container">
      <form onSubmit={handleSubmit}>
        <h3>Login</h3>

        <input
          placeholder="Username"
          id="username"
          name="username"
          value={formData.username}
          onChange={handleChange}
        />

        <input
          placeholder="Password"
          type="password"
          id="password"
          name="password"
          value={formData.password}
          onChange={handleChange}
        />

        {formErrors.map((e) => (
          <p className="error">{e}</p>
        ))}
        <button>Login</button>
      </form>
    </div>
  );
};

export default LoginForm;
