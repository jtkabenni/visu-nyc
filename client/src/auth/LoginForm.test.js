import React from "react";
import { render } from "@testing-library/react";
import LoginForm from "./LoginForm";
import { MemoryRouter } from "react-router";

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <LoginForm />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
