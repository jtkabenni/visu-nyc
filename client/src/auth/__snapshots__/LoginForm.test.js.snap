// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="form-container"
  >
    <form>
      <h3>
        Login
      </h3>
      <input
        id="username"
        name="username"
        placeholder="Username"
        value=""
      />
      <input
        id="password"
        name="password"
        placeholder="Password"
        type="password"
        value=""
      />
      <button>
        Login
      </button>
    </form>
  </div>
</DocumentFragment>
`;
